src/main/java/org/prebid/server/bidder/adnuntius/AdnuntiusBidder.java

```diff
@@ -0,0 +1,252 @@
+import org.prebid.server.bidder.model.BidderBid;
+import org.prebid.server.bidder.model.BidderError;
+import org.prebid.server.bidder.model.HttpCall;
+import org.prebid.server.bidder.model.HttpRequest;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.json.DecodeException;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.request.ExtRegs;
+import org.prebid.server.proto.openrtb.ext.request.ExtUser;
+import org.prebid.server.proto.openrtb.ext.request.adnuntius.ExtImpAdnuntius;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
+import org.prebid.server.util.ObjectUtil;
+
+import java.math.BigDecimal;
+import java.net.URISyntaxException;
+import java.time.Clock;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class AdnuntiusBidder implements Bidder<AdnuntiusRequest> {
+
+    private static final int SECONDS_IN_MINUTE = 60;
+    private static final String TARGET_ID_DELIMITER = "-";
+    private static final String DEFAULT_PAGE = "unknown";
+    private static final BigDecimal PRICE_MULTIPLIER = BigDecimal.valueOf(1000);
+    private static final TypeReference<ExtPrebid<?, ExtImpAdnuntius>> ADNUNTIUS_EXT_TYPE_REFERENCE =
+            new TypeReference<>() {
+            };
+
+    private final String endpointUrl;
+    private final Clock clock;
+    private final JacksonMapper mapper;
+
+    public AdnuntiusBidder(String endpointUrl, Clock clock, JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.clock = Objects.requireNonNull(clock);
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<AdnuntiusRequest>>> makeHttpRequests(BidRequest request) {
+        final Map<String, List<AdnuntiusAdUnit>> networkToAdUnits = new HashMap<>();
+        final List<HttpRequest<AdnuntiusRequest>> adnuntiusRequests = new ArrayList<>();
+
+        try {
+            for (Imp imp : request.getImp()) {
+                validateImp(imp);
+
+                final ExtImpAdnuntius extImpAdnuntius = parseImpExt(imp);
+
+                final String key = StringUtils.stripToEmpty(extImpAdnuntius.getNetwork());
+                final String auId = extImpAdnuntius.getAuId();
+                networkToAdUnits.computeIfAbsent(key, k -> new ArrayList<>())
+                        .add(AdnuntiusAdUnit.of(auId, auId + TARGET_ID_DELIMITER + imp.getId()));
+            }
+        } catch (PreBidException e) {
+            return Result.withError(BidderError.badInput(e.getMessage()));
+        }
```

Comment:
Lets narrow down try-catch scope. only call to `parseImpExt` should be surrounded with try-catch.